examples/ASMapNode/Sample/MapHandlerNode.m

```diff
@@ -90,6 +91,25 @@ - (void)didLoad
   [_liveMapToggleButton setTitle:[self liveMapStr] withFont:nil withColor:[UIColor blueColor] forState:ASControlStateNormal];
   [_liveMapToggleButton setTitle:[self liveMapStr] withFont:[UIFont systemFontOfSize:14] withColor:[UIColor blueColor] forState:ASControlStateHighlighted];
   [_liveMapToggleButton addTarget:self action:@selector(toggleLiveMap) forControlEvents:ASControlNodeEventTouchUpInside];
+  
+  // avoiding retain cycles
+  __weak MapHandlerNode *weakSelf = self;
+  
+  self.mapNode.imageForStaticMapAnnotationBlock = ^UIImage *(id<MKAnnotation> annotation, CGPoint *centerOffset){
+    MapHandlerNode *grabbedSelf = weakSelf;
+    if (grabbedSelf) {
+      MKAnnotationView *av = [grabbedSelf annotationViewForAnnotation:annotation];
+      *centerOffset = av.centerOffset;
+      UIGraphicsBeginImageContextWithOptions(av.bounds.size, av.opaque, 0.0);
+      [av.layer renderInContext:UIGraphicsGetCurrentContext()];
+      UIImage *pinImage = UIGraphicsGetImageFromCurrentImageContext();
```

Comment:
Since this block is going to be run on an arbitrary serial queue, we can render any UIViews or CALayers here. Could you check if the annotation is a `CustomMapAnnotation` and return its image directly? For `MKPinAnnotation`'s default image, maybe the best thing is to add something like this on line 97:

```objectivec
static UIImage *pinImage;
static dispatch_once_t onceToken;
dispatch_once(&onceToken, ^{
    pinImage = [[[MKPinAnnotationView alloc] init] image];
});
```

That way, on the main thread, you read the pin image one time, and you can return it from that block safely.